server/route.go

```diff
@@ -36,6 +36,14 @@ func (srv *Server) initHandlers() {
 		c.Set(consts.ContextKeyStore, srv.store)
 		c.Next()
 	}, middleware.RedirectIfNotLeader)
+
+	if srv.config.ApiAuth.ApiUser != "" && srv.config.ApiAuth.ApiPassword != "" {
```

Comment:
This condition can be extracted in a single method of config like config.AuthConfigured().